Gems/Atom/RHI/Vulkan/Code/Source/RHI/Device.cpp

```diff
@@ -31,6 +31,13 @@
 #include <Vulkan_Traits_Platform.h>
 #include <Atom/RHI.Reflect/VkAllocator.h>
 
+#define VMA_IMPLEMENTATION
+#define VMA_STATIC_VULKAN_FUNCTIONS 0
+
+#pragma warning(push, 3)
+#include <vma/vk_mem_alloc.h>
+#pragma warning(pop)
```

Comment:
This should use the [AZ_PUSH_DISABLE_WARNING/AZ_POP_DISABLE_WARNING](https://github.com/o3de/o3de/blob/a10351f38ddf50af2b01ed071c04cda52d6069ed/Code/Framework/AzCore/AzCore/PlatformDef.h#L73-L80) macros.

This will not work on non-Windows.

Reply:
removed the pragma's